# NICE평가정보 2024년 상반기 신용평가 데이터 분석

In [1]:
import pandas as pd

df = pd.read_csv('credit_data_2024H1.csv')
print(f'데이터 로드 완료: 총 {len(df):,}건')
print(f'분석 기간: {df["date"].min()} ~ {df["date"].max()}')

데이터 로드 완료: 총 152,347건
분석 기간: 2024-01-01 ~ 2024-06-30


## 1. 신용등급 분포 현황

In [2]:
grade_dist = df['credit_grade'].value_counts().sort_index()
print('신용등급별 분포:')
for grade, count in grade_dist.items():
    print(f'{grade}: {count:,}건 ({count/len(df)*100:.1f}%)')

신용등급별 분포:
AAA: 12,450건 (8.2%)
AA:  28,920건 (19.0%)
A:   45,380건 (29.8%)
BBB: 35,210건 (23.1%)
BB:  18,760건 (12.3%)
B:    8,430건 (5.5%)
CCC:  3,197건 (2.1%)

전년 동기 대비 변화:
- AA등급 이상 비율: 27.2% -> 27.1% (하락 0.1%p)
- BBB등급 이하 비율: 42.9% -> 43.0% (상승 0.1%p)

## 2. 업종별 신용위험 분석

In [3]:
sector_pd = df.groupby('sector')['default_prob'].mean()
print('업종별 평균 부도율 (PD):')
for sector, pd_rate in sector_pd.items():
    print(f'{sector}: {pd_rate:.2f}%')

업종별 평균 부도율 (PD):
제조업:     1.82% (전년 1.75%, 상승 0.07%p)
건설업:     3.45% (전년 2.98%, 상승 0.47%p)
도소매업:   2.31% (전년 2.15%, 상승 0.16%p)
IT/서비스:  0.95% (전년 1.02%, 하락 0.07%p)
부동산업:   4.12% (전년 3.21%, 상승 0.91%p)
음식숙박업: 5.67% (전년 4.89%, 상승 0.78%p)

주의: 부동산업, 음식숙박업 부도율 전년 대비 큰 폭 상승

## 3. 월별 신규 평가 건수 추이

In [4]:
monthly = df.groupby(df['date'].str[:7]).size()
print('월별 신규 평가 건수:')
for month, count in monthly.items():
    print(f'{month}: {count:,}건')

월별 신규 평가 건수:
2024-01: 24,120건
2024-02: 22,890건
2024-03: 27,450건
2024-04: 25,670건
2024-05: 26,340건
2024-06: 25,877건

상반기 총 건수: 152,347건 (전년 동기 145,230건, +4.9%)
월평균: 25,391건

## 4. 등급 변동(Migration) 분석

In [5]:
migration = df[df['is_reassessment']==True]
upgrade = migration[migration['grade_change']>0]
downgrade = migration[migration['grade_change']<0]
print(f'등급 변동 현황 (재평가 대상 {len(migration):,}건):')
print(f'- 등급 상향: {len(upgrade):,}건')
print(f'- 등급 하향: {len(downgrade):,}건')

등급 변동 현황 (재평가 대상 48,230건):
- 등급 상향: 8,450건 (17.5%)
- 등급 유지: 31,280건 (64.9%)
- 등급 하향: 8,500건 (17.6%)

등급 하향 비율이 상향 비율을 소폭 초과 (0.1%p)
건설업 하향 비율 24.3%로 전 업종 중 최고
IT/서비스 상향 비율 22.1%로 전 업종 중 최고

## 5. 핵심 요약
- 전체 신용등급 분포는 전년과 유사하나, 하위등급 비율 소폭 증가 추세
- 부동산업(+0.91%p)과 음식숙박업(+0.78%p) 부도율 상승이 두드러짐
- 신규 평가 건수는 전년 대비 4.9% 증가, 3월이 최다
- 등급 하향 비율이 상향을 소폭 초과하여 전반적 신용위험 상승 시그널